src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -101,6 +115,7 @@ internal void Walk(IList<OpenApiServer> servers)
                     Walk(server);
                 }
```

Comment:
We should add the array index to get the correct json pointer here.

                for ( int i =0; i<servers.Count(); i++ )
                {
                    _visitor.Enter(i);
                    Walk(servers[i]);
                     _visitor.Exit();
                }